src/ibex_bluesky_core/callbacks/plotting.py

```diff
@@ -74,3 +79,24 @@ def start(self, doc: RunStart) -> None:
         """Process an start document (delegate to superclass, then show the plot)."""
         super().start(doc)
         show_plot()
+
+
+class PlotPNGSaver(CallbackBase):
```

Comment:
You might need to make this derive from `QtAwareCallback` if you want to avoid race conditions where you try to save the plot before it's been drawn (since the actual plotting is done in a QtAware, and therefore potentially on other threads, meaning it hasn't necessarily finished by the time you get here